sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/ReadTimeoutHandler.java

```diff
@@ -65,6 +66,10 @@ private void readTimeoutRunnable(ChannelHandlerContext ctx) {
         }
 
         // No progress has been made since the last timeout event, channel has timed out.
-        ctx.fireExceptionCaught(new TimeoutException(String.format(READ_TIMED_OUT_MESSAGE, timeoutMillis)));
+        if (!closed) {
```

Comment:
Why is this check required? Can this method be called multiple times? If yes, what's the impact of trying to close the context each time?

Reply:
The check is there to ensure that we don't fire the exception multiple times. As for what happens if close is called multiple times I am not certain and can take a look into this.

Reply:
Looked into the safety of calling close multiple times and the answer is unknown as this will trigger the close handler for other hanndlers, so it would be dependent on the configuration of the Netty pipeline. In our default instance it is safe to call multiple times but that isn't a guarantee.